clap_complete/src/dynamic/completer.rs

```diff
@@ -64,18 +65,22 @@ pub fn complete(
             if let Some(next_cmd) = current_cmd.find_subcommand(value) {
                 current_cmd = next_cmd;
                 pos_index = 1;
+                state = ParseState::ValueDone;
                 continue;
             }
         }
 
         if is_escaped {
             pos_index += 1;
+            state = ParseState::Pos(pos_index);
         } else if arg.is_escape() {
             is_escaped = true;
+            state = ParseState::ValueDone;
         } else if let Some(_long) = arg.to_long() {
         } else if let Some(_short) = arg.to_short() {
         } else {
             pos_index += 1;
+            state = ParseState::Pos(pos_index);
         }
```

Comment:
The current value being positional shouldn't make the next one positional.

Instead we still need to pass `pos_index` to `complete_arg`.  We can only drop `is_escaped`

Reply:
Yes, this is a logical error. Here it should be `ParseState::ValueDone` rather than `ParseState::Pos`. And `pos_index` is like a global context during parsing, `ParseState::Pos(usize)` should only be just useful after `--`.

Reply:
Later, it'll be useful for `num_args`